### PR TITLE
Environment variables to control MLIR execution

### PR DESCRIPTION
**Pull request by slyalin**
Introducing a few environment variables to control MLIR stack in run-time. All environment variables have the prefix `OV_MLIR_` and are enabled/disabled by setting values `1`, `true`, `ON` and `0`, `false`, `OFF` correspondingly. Values are case insensitive.

- `OV_MLIR`: Enables MLIR stack. When disabled the execution is the same as in vanilla OpenVINO. Default: `ON`.
- `OV_MLIR_TPP`: Enables TPP-MLIR. OpenVINO should be built with TPP-MLIR support to be enabled by this variable. Default: `ON` if built with TPP-MLIR, `OFF` if built without TPP-MLIR. If OpenVINO is built without TPP-MLIR and the variable is set to enable it, a run-time exception will be thrown.
- `OV_MLIR_DEBUG`: Enables debug output in MLIR stack. Default: `OFF`. Earlier it was always enabled, now it should be enabled explicitly.

### CODE REVIEW

**Comment by slyalin** [comment]
@adam-smnk, you will find it useful to run benchmarks without rebuilds and compare everything: OV vs. MLIR upstream vs. TPP-MLIR.

File: src/common/transformations/src/transformations/mlir/op/matmul.cpp [hunk 1]
```diff
@@ -50,7 +44,16 @@ using namespace ov::pass::pattern;
 using namespace ov::op;
 
 MatMulPattern::MatMulPattern() : MarkPattern(
-    wrap_type<v0::MatMul>({any_input(), any_input()}),
+    wrap_type<v0::MatMul>({any_input(), any_input()}, [](const Output<Node>& output) {
```
Comment (adam-smnk) [inline src/common/transformations/src/transformations/mlir/op/matmul.cpp]: I guess not rebased on `mlir` branch?

Comment (slyalin) [reply src/common/transformations/src/transformations/mlir/op/matmul.cpp]: Yes, just doing that.

**Review by adam-smnk — APPROVED** [review]
Looks good, thanks

**Pull request merged**